pkg/networkservice/common/authorize/server.go

```diff
@@ -28,19 +31,55 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
 
-type authorizeServer struct{}
+type authorizeServer struct {
+	policy *rego.PreparedEvalQuery
+}
 
 // NewServer - returns a new authorization networkservicemesh.NetworkServiceServers
-func NewServer() networkservice.NetworkServiceServer {
-	return &authorizeServer{}
+func NewServer(p *rego.PreparedEvalQuery) networkservice.NetworkServiceServer {
+	return &authorizeServer{
+		policy: p,
+	}
 }
 
 func (a *authorizeServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
-	// TODO check authorization
+	rs, err := a.policy.Eval(ctx, rego.EvalInput(request.GetConnection().GetPath().GetPathSegments()[request.GetConnection().GetPath().GetIndex()].GetToken()))
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	if err := check(rs); err != nil {
+		return nil, err
+	}
+
 	return next.Server(ctx).Request(ctx, request)
 }
 
 func (a *authorizeServer) Close(ctx context.Context, conn *connection.Connection) (*empty.Empty, error) {
-	// TODO check authorization
+	rs, err := a.policy.Eval(ctx, rego.EvalInput(conn.GetPath().GetPathSegments()[conn.GetPath().GetIndex()].GetToken()))
```

Comment:
Are we only considering the current token?  Would it be possible to reason about more than that?

Reply:
We can pass here the whole `Request` or `Path`. I think we can do that after adding some real-world policy (not the test one)

Reply:
#51 has been created and added to the project board. Resolving this to unblock merging.